File: java/behavioral_pattern/observer/CurrentConditionsDisplay.java
package observer;

public class CurrentConditionsDisplay implements Observer {
    private float temperature;
    private float humidity;

    @Override
    public void update(float temperature, float humidity, float pressure) {
        this.temperature = temperature;
        this.humidity = humidity;

    }

    public void display() {
        System.out.println("Current conditions: " + temperature + " °C and " + humidity + "%");
    }
}

File: typescript/behavioral_pattern/src/observer/CurrentConditionsDisplay.ts
import { Observer } from "./Observer";

class CurrentConditionsDisplay implements Observer {
    private temperature: number = 0;
    private humidity: number = 0;

    public update(temperature: number, humidity: number, pressure: number): void {
        this.temperature = temperature;
        this.humidity = humidity;
    }

    public display(): void {
        console.log("Current Conditions: " + this.temperature + " °C and " + this.humidity + "%");
    }
}

export default CurrentConditionsDisplay;